File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/browsing/items/media/files/cached/stream/supplier/ICacheStreamSupplier.java
package com.lasthopesoftware.bluewater.client.browsing.items.media.files.cached.stream.supplier;

import com.lasthopesoftware.bluewater.client.browsing.items.media.files.cached.stream.CacheOutputStream;
import com.namehillsoftware.handoff.promises.Promise;

public interface ICacheStreamSupplier {
	Promise<CacheOutputStream> promiseCachedFileOutputStream(final String uniqueKey);
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/browsing/items/media/files/cached/stream/supplier/ICacheStreamSupplier.kt
package com.lasthopesoftware.bluewater.client.browsing.items.media.files.cached.stream.supplier

import com.lasthopesoftware.bluewater.client.browsing.items.media.files.cached.stream.CacheOutputStream
import com.namehillsoftware.handoff.promises.Promise

interface ICacheStreamSupplier {
    fun promiseCachedFileOutputStream(uniqueKey: String): Promise<CacheOutputStream>
}